mutable_tree.go

```diff
@@ -310,6 +312,25 @@ func (tree *MutableTree) LoadVersion(targetVersion int64) (int64, error) {
 	return latestVersion, nil
 }
 
+func (tree *MutableTree) LatestVersion(maxVersion int64) (int64, error) {
+	roots, err := tree.ndb.getRoots()
+	if err != nil {
+		return 0, err
+	}
+
+	if len(roots) == 0 {
+		return 0, nil
+	}
+
+	latestVersion := int64(0)
+	for version := range roots {
```

Comment:
range map is randomized and it is not guaranteed that it will return sorted values. We need to put all version into array sort it and then select the latest version out of the array.